### PR TITLE
persistent reservation address left comments

### PR DESCRIPTION
**Pull request by alicefr**
**What this PR does / why we need it**:
This PR address some left comments from #9177 and added unit test for the socket device plugin

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/device-manager/socket_device_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		Expect(<-errChan).ToNot(HaveOccurred())
+	})
+
+	It("Should monitor health of device node", func() {
+		os.OpenFile(dpi.pluginSocketPath, os.O_RDONLY|os.O_CREATE, 0666)
+
+		go dpi.healthCheck()
+		Expect(dpi.devs[0].Health).To(Equal(pluginapi.Healthy))
+
+		time.Sleep(1 * time.Second)
```
Comment (xpivarc) [inline pkg/virt-handler/device-manager/socket_device_test.go:64]: Is this required?

**Review by xpivarc — COMMENTED** [review]
/approve
Few minor nits, see below

File: pkg/virt-handler/device-manager/socket_device_test.go (hunk 1 above)
Comment (alicefr) [reply pkg/virt-handler/device-manager/socket_device_test.go:64]: Mh, this comes from https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/device-manager/generic_device_test.go#L69, and it is for giving a little bit of time to the go routine to start. I'll refactor a bit.

File: tests/storage/reservation.go [hunk 2]
```diff
@@ -191,7 +189,7 @@ var _ = SIGDescribe("[Serial]SCSI persistent reservation", Serial, func() {
 		}
 	})
 
-	Context("Use LUN disk with presistent reservation", func() {
+	FContext("Use LUN disk with presistent reservation", func() {
```
[the remark this answers is not in the document]
Comment (alicefr) [reply tests/storage/reservation.go]: Thanks!

File: pkg/virt-handler/device-manager/socket_device_test.go (hunk 1 above)
Comment (alicefr) [reply pkg/virt-handler/device-manager/socket_device_test.go]: Ok, in the end this isn't necessary. The healthCheck needs to work in the same way also if th socket has already been removed before the routine starts

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by alicefr** [comment]
@xpivarc do you mind adding a lgtm to this pr?

**Comment by alicefr** [comment]
Rebased

File: pkg/virt-handler/device-manager/socket_device.go [hunk 3]
```diff
@@ -258,13 +259,10 @@ func (dpi *SocketDevicePlugin) healthCheck() error {
 	}
 	defer watcher.Close()
 
-	// This way we don't have to mount /dev from the node
```
Comment (lyarwood) [inline pkg/virt-handler/device-manager/socket_device.go:261]: nit - unrelated comment removal?

**Review by lyarwood — APPROVED** [review]
/lgtm

supernit around an unrelated comment you removed in the second commit but I'm not sure it's that useful to warrant holding this?

File: pkg/virt-handler/device-manager/socket_device.go (hunk 3 above)
Comment (alicefr) [reply pkg/virt-handler/device-manager/socket_device.go:261]: It was a leftover from a copy-paste from another device plugin. We don't use any /dev with the socket plugin

**Pull request merged**